service/src/main/java/bio/terra/workspace/db/ApplicationDao.java

```diff
@@ -179,6 +192,19 @@ public WsmWorkspaceApplication enableWorkspaceApplication(
     return enableWorkspaceApplicationWorker(workspaceUuid, applicationId);
   }
 
+  /**
+   * Enable a list of applications. Used as part of create workspace
+   *
+   * @param workspaceUuid workspace to enable applications in
+   * @param applicationIds list of application ids
+   */
+  @WriteTransaction
+  public void enableWorkspaceApplications(UUID workspaceUuid, List<String> applicationIds) {
+    for (String applicationId : applicationIds) {
+      enableWorkspaceApplication(workspaceUuid, applicationId);
```

Comment:
How do transactions work if this method is annotated `@WriteTransaction` and `enableWorkspaceApplication` is also `@WriteTransaction`? Are they nested inside one larger transaction, or does each application end up in its own transaction?

Reply:
`@___Transaction` annotations are set to propagate-always. The transaction manager layer sees the existing outer transaction and uses it instead of starting a new transaction.